src/arch/metal.rs

```diff
@@ -809,6 +834,12 @@ __bss_stop:
 // https://github.com/rust-lang/rust/issues/93332
 static STACK_SIZE: usize = TARGET_STACK_SIZE;
 
+// Boot hart ID
+static BOOT_HART_ID: usize = PLATFORM_BOOT_HART_ID;
+
+// Boolean to synchronized harts
+static BOOT_BSS_SET: bool = true;
```

Comment:
Rust does not give layout guarantees for booleans. So this code might break when upgrading the compiler if the layout changes for some reason. Also big/little endian change the conversions between boolean and integers.

So you should pick a well defined integer size and initialize it with a proper layout. Then be sure that your load matches the size of the integer type you picked.

Reply:
Has `usize` a well-defined integer size?

Reply:
It is platform dependent (32 or 64 bits), but yes the layout is specified ^^